components/remerge/src/storage/bootstrap.rs

```diff
@@ -114,3 +115,69 @@ pub(super) fn bootstrap(
         guid,
     ))
 }
+
+fn get_schema(
+    db: &Connection,
+    schema_ver: &String
+) -> Result<RecordSchema, SchemaError> {
+    let native: String = db.query_row(
+        "SELECT schema_text FROM remerge_schemas WHERE version = ?",
+        rusqlite::params![schema_ver],
+        |r| r.get(0),
+    )
+    .unwrap();
+
+    crate::schema::parse_from_string(&*native, false)
+}
+
+fn migrate_records(
+    db: &Connection,
+    previous_native: &SchemaBundle,
+    // local: &RecordSchema
+    new_native: &SchemaBundle
+) -> Result<()> {
+    // Check if there are dedupe_on fields
+    if !new_native.local.dedupe_on.is_empty() {
+        unimplemented!("FIXME: migration");
+    }
+
+    // Get existing records with old schema version
+    let mut stmt = db.prepare_cached("SELECT record_data FROM rec_mirror WHERE is_overridden = 0")?;
+    let rows = stmt.query_and_then(rusqlite::NO_PARAMS, |row| -> Result<NativeRecord> {
+        let r: LocalRecord = row.get("record_data")?;
+        previous_native.local_to_native(&r)
+    })?;
+    let old_records: Vec<NativeRecord> = rows.collect::<Result<_>>().unwrap_or_default();
+    // let mut new_records: Vec<NativeRecord> = Vec::new();
+
+    for old_record in old_records {
+        let mut new_record_data = crate::JsonObject::default();
+
+        for new_record_field in &new_native.local.fields {
+            let value = old_record.as_obj()[new_record_field.name.as_str()].clone();
+
+            //  Apply defaults from the new schema where applicable.
+            new_record_data.insert(new_record_field.clone().name, value);
+        }
+        // &new_records.push(NativeRecord::new_unchecked(new_record_data));
+        let new_native_record = NativeRecord::new_unchecked(new_record_data);
+        let (id, record) = new_native.native_to_local(&new_native_record, ToLocalReason::Creation)?;
```

Comment:
@thomcc I really don't know how sane this code is. I may be dabbling at the edge of my Rust knowledge but I need functionality that's in [the storage create function](https://github.com/mozilla/application-services/blob/ec0d019d1004269ad7921b8230e50741541eaf63/components/remerge/src/storage/db.rs#L105), and the only way I know how to access that functionality would be to create an instance of RemergeDB, which seems like a very bad idea. But copying the logic from the create function (as I was starting to do above) seems less than ideal. Is there a better way to do this that I'm missing?